app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,17 @@
 # Displays the admins' homepage
 class AdminController < ApplicationController
+  before_action :authorize
+
   def index
   end
+
+
+  protected
+
+  def authorize
+    unless User.find_by(id: session[:user_id])
+      redirect_to login_url, notice: 'Please log in to access this page'
+    end
+  end
```

Comment:
In general I like to isolate the lookup of models from params. I think you've already done this for book, language, etc. in other controllers. Doing it for the user is just following this pattern.

And yeah, it can be useful to have it generally available.